From 6dc14e91fe117c628e380c2d4a5672e3f7d9e04b Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Mon, 14 Nov 2022 07:19:48 -0800
Subject: [PATCH 184/196] Fix `update-chainparams.sh` so it can be ran on MacOS
 X (#294)

Script now looks for an env var called `CONFIG_FILE` that will override
the default config location. Can be ran as:

```
CONFIG_FILE=CONFIG_FILE=~/Library/Application\ Support/Lotus/lotus.conf contrib/source-control-tools/patch-recipes/update-chainparams.sh
```
---
 .../patch-recipes/update-chainparams.sh                  | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/contrib/source-control-tools/patch-recipes/update-chainparams.sh b/contrib/source-control-tools/patch-recipes/update-chainparams.sh
index a39fcf7f9..885a22f8f 100755
--- a/contrib/source-control-tools/patch-recipes/update-chainparams.sh
+++ b/contrib/source-control-tools/patch-recipes/update-chainparams.sh
@@ -23,13 +23,18 @@ while [ "${INDEX}" -lt "${#ARGS[@]}" ]; do
   ((INDEX+=1))
   TESTNET_PORT="${ARGS[${INDEX}]}"
   ((INDEX+=1))
+  if [[ -z "$CONFIG_FILE" ]]; then
+    CONFIG_FLAG="--config=~/.lotus/lotus.conf"
+  else
+    CONFIG_FLAG="--config=${CONFIG_FILE}"
+  fi
 
   CHAINPARAMS_MAINNET_TXT="${NETWORK}/chainparams_main.txt"
-  ./make_chainparams.py -a 127.0.0.1:"${MAINNET_PORT}"> "${CHAINPARAMS_MAINNET_TXT}"
+  ./make_chainparams.py -a 127.0.0.1:"${MAINNET_PORT}" "${CONFIG_FLAG}" > "${CHAINPARAMS_MAINNET_TXT}"
   git add "${CHAINPARAMS_MAINNET_TXT}"
 
   CHAINPARAMS_TESTNET_TXT="${NETWORK}/chainparams_test.txt"
-  ./make_chainparams.py -a 127.0.0.1:"${TESTNET_PORT}" > "${CHAINPARAMS_TESTNET_TXT}"
+  ./make_chainparams.py -a 127.0.0.1:"${TESTNET_PORT}" "${CONFIG_FLAG}" > "${CHAINPARAMS_TESTNET_TXT}"
   git add "${CHAINPARAMS_TESTNET_TXT}"
 
   CHAINPARAMS_CONSTANTS="${TOPLEVEL}"/src/networks/"${NETWORK}"/chainparamsconstants.cpp
-- 
2.39.5 (Apple Git-154)

